src/deploy.rs

```diff
@@ -1,31 +1,100 @@
+use std::array::TryFromSliceError;
+use std::num::ParseIntError;
 use std::{fmt::Debug, fs, io};
 
 use clap::Parser;
+use ed25519_dalek::Signer;
 use hex::FromHexError;
-use soroban_env_host::{xdr::Error as XdrError, HostError};
+use jsonrpsee_core::client::ClientT;
+use jsonrpsee_core::rpc_params;
+use jsonrpsee_http_client::HttpClientBuilder;
+use rand::Rng;
+use sha2::{Digest, Sha256};
+use soroban_env_host::xdr::{
+    DecoratedSignature, Error as XdrError, Hash, HashIdPreimage, HashIdPreimageEd25519ContractId,
+    HostFunction, InvokeHostFunctionOp, LedgerFootprint, LedgerKey::ContractData,
+    LedgerKeyContractData, Memo, MuxedAccount, Operation, OperationBody, Preconditions, ScObject,
+    ScStatic::LedgerKeyContractCode, ScVal, SequenceNumber, Signature, SignatureHint, Transaction,
+    TransactionEnvelope, TransactionExt, TransactionSignaturePayload,
+    TransactionSignaturePayloadTaggedTransaction, TransactionV1Envelope, Uint256, VecM, WriteXdr,
+};
+use soroban_env_host::HostError;
+use stellar_strkey::StrkeyPrivateKeyEd25519;
 
 use crate::snapshot::{self, get_default_ledger_info};
 use crate::utils;
 
 #[derive(Parser, Debug)]
 pub struct Cmd {
-    #[clap(long = "id")]
-    /// Contract ID to deploy to
-    contract_id: String,
     /// WASM file to deploy
     #[clap(long, parse(from_os_str))]
     wasm: std::path::PathBuf,
-    /// File to persist ledger state
-    #[clap(long, parse(from_os_str), default_value(".soroban/ledger.json"))]
+    #[clap(
+        long = "id",
+        required_unless_present = "rpc-server-url",
+        conflicts_with = "rpc-server-url"
+    )]
+    // TODO: Should we get rid of the contract_id parameter
+    //       and just obtain it from the key/source like we do
+    //       when running against an rpc server?
```

Comment:
I understand but shouldn't we strive to have arguments as similar as possible? (so that moving to the network from the sandbox becomes as smooth as possible)

We can discuss this later though.

Reply:
It's inconvenient to require a key and source when hacking on a contract and wanting to see it deployed in sandbox. It's much easier to just say it should have ID `0x1`. But I agree there's a tradeoff here with similarity to a real deployment. I'll play around with the broad ergonomics next week once we're no longer focused on the functional side of things, so I suggest we defer this for the moment.

Reply:
OK, let's leave it as it is for now.